test/Microsoft.AspNet.Proxy.Test/PassthroughProxyTest.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.TestHost;
+using Microsoft.Net.Http.Server;
+using Xunit;
+
+namespace Microsoft.AspNet.Proxy.Test
+{
+    public class PassthroughProxyTest
+    {
+        [Fact]
+        public async Task PassthoughGetRequest()
+        {
+            var port = "3001";
+            var methodType = "GET";
+            var options = new ProxyOptions()
+            {
+                Host = "localhost",
+                Scheme = "http",
+                Port = port
+            };
+
+            var req = new HttpRequestMessage(HttpMethod.Get, "");
+            var response = await SendRequestAndRecieveRequest(req, options, port, methodType);
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal("OK", response.ReasonPhrase);
+            Assert.Equal(new Version(1, 1), response.Version);
+        }
+
+        [Fact]
+        public async Task PassthroughPostRequestNoBody()
+        {
+            var port = "3002";
+            var MethodType = "POST";
+            var options = new ProxyOptions()
+            {
+                Host = "localhost",
+                Scheme = "http",
+                Port =  port
+            };
+
+            var req = new HttpRequestMessage(HttpMethod.Post, "");
+            var response = await SendRequestAndRecieveRequest(req, options, port, MethodType);
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal("OK", response.ReasonPhrase);
+            Assert.Equal(new Version(1, 1), response.Version);
+        }
+
+        private async Task<HttpResponseMessage> SendRequestAndRecieveRequest(HttpRequestMessage req, ProxyOptions options, String port, String methodType)
+        {
+            var waitForWebListener = new ManualResetEvent(false);
+            var keepServerAlive = new ManualResetEvent(false);
+            var returnResponse = new ManualResetEvent(false);
+
+            var thread = new Thread(async () =>
+            {
+                using (WebListener listener = new WebListener())
```

Comment:
You shouldn't use WebListener in unit tests. Whatever happened to that discussion around Backchannels?